### PR TITLE
chore(js): upgrade react to 17.0.1

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

Upgrades React to version 17.0.1 for the monorepo and upgrades various associated packages:

 - react-dom
 - @hot-loader/react-dom
 - react-hot-loader: latest version is react 17-compatible but deprecated (see https://github.com/Opentrons/opentrons/pull/7990).
 - @types/react
 - react-test-renderer
 - @wojtekmaj/enzyme-adapter-react-17: an unofficial but widely used enzyme adapter for react 17
 - react-remove-scroll

Upgrades to 17.0.1 instead of latest (17.0.2) because @hot-loader/react-dom is only up to 17.0.1 as of now

Also a few TS error fixes, some React 16-related workarounds removed, and some viewport/modal changes to the protocol designer cypress tests to hopefully make them less flaky

Closes https://github.com/Opentrons/opentrons/issues/8535

# Changelog

 - Upgrades React to version 17.0.1

# Review requests

 - Smoke test js projects

# Risk assessment

Moderate: affects every react project, could be unexpected problems

### CODE REVIEW

**Review by smb2268 — APPROVED** [review]
🚀

**Review by b-cooper — APPROVED** [review]
I didn't see any smoke.  Looks good 👍

File: protocol-designer/cypress/support/commands.js [hunk 1]
```diff
@@ -29,7 +29,12 @@ import 'cypress-file-upload'
 // General Custom Commands
 //
 Cypress.Commands.add('closeAnnouncementModal', () => {
-  cy.get('button').contains('Got It!').click()
+  // ComputingSpinner sometimes covers the announcement modal button and prevents the button click
+  // this will retry until the ComputingSpinner does not exist
+  cy.get('[data-test="ComputingSpinner"]', { timeout: 30000 }).should(
+    'not.exist'
+  )
+  cy.get('button').contains('Got It!').should('be.visible').click()
```
Comment (brenthagen) [inline protocol-designer/cypress/support/commands.js:37]: @reachojhaneha do you see any issues with this change?  I saw that the computing spinner needs this generous timeout in other places.

Comment (reachojhaneha) [reply protocol-designer/cypress/support/commands.js:37]: the cypress changes looks good to me

**Pull request merged**